docs/udf.rst

```diff
@@ -0,0 +1,179 @@
+
++------------+------------------------------+---------+
+| Field      | Description                  | Required|
++============+==============================+=========+
+| name       | The name of the function(s)  | Y       |
+|            | represented by this class.   |         |
++------------+------------------------------+---------+
+| description| A string describing generally| Y       |
+|            | what the function(s) in this |         |
+|            | class do.                    |         |
++------------+------------------------------+---------+
+| author     | The author of the UDF.       | N       |
++------------+------------------------------+---------+
+| version    | The version of the UDF.      | N       |
++------------+------------------------------+---------+
+
+
+Udf Annotation
+--------------
+
+The ``@Udf`` annotation is applied to public methods of a class annotated with ``@UdfDescription``.
+Each annotated method will become an invocable function in KSQL. The annotation only has a single
+field ``description`` that is required. You can use this to better describe what a particular version
+of the UDF does, for example:
+
+.. code:: java
+
+    @Udf(description = "Returns a string that is a substring of this string. The"
+        + " substring begins with the character at the specified startIndex and"
+        + " extends to the end of this string.")
+    public String substring(final String value, final int startIndex)
+
+    @Udf(description = "Returns a string that is a substring of this string. The"
+        + " substring begins with the character at the specified startIndex and"
+        + " extends to the character at endIndex -1.")
+    public String substring(final String value, final int startIndex, final int endIndex)
+
+
+Example UDF class
+-----------------
+
+The class below creates a UDF named ``multiply``. As can be seen this UDF can be invoked in 3
+different ways:
+
+- with two int parameters returning a long result.
+- with two long parameters returning a long result.
+- with two double parameters returning a double result.
+
+.. code:: java
+
+    @UdfDescription(name = "multiply", description = "multiplies 2 numbers")
+    public class Multiply {
+
+      @Udf(description = "multiply two ints")
+      public long multiply(final int v1, final int v2) {
+        return v1 * v2;
+      }
+
+      @Udf(description = "multiply two longs")
+      public long multiply(final long v1, final long v2) {
+        return v1 * v2;
+      }
+
+      @Udf(description = "multiply two doubles")
+      public double multiply(final double v1, double v2) {
+        return v1 * v2;
+      }
+
+    }
+
+Supported Types
+---------------
+
+The types supported by UDFs are currently limited to:
+
+- int, Integer
+- boolean, Boolean
+- long, Long
+- double, Double
+- String
+- java.util.Map (must be a map of supported types)
+- java.util.List (must be a list of supported types)
+
```

Comment:
Maybe have these in a table with their corresponding type in KSQL. This would help user to know what to pass as argument and what to expect as return. For instance it's better to clarify List is `ARRAY` in KSQL.

Reply:
Yeah, even better. Good point @hjafarpour